www/js/diary/list/TimelineScrollList.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { FlashList } from '@shopify/flash-list';
+import { array, func, object, oneOfType, bool, string } from 'prop-types';
+import TripCard from '../cards/TripCard';
+import PlaceCard from '../cards/PlaceCard';
+import UntrackedTimeCard from '../cards/UntrackedTimeCard';
+import { View } from 'react-native';
+import { ActivityIndicator } from 'react-native-paper';
+import LoadMoreButton from './LoadMoreButton';
+
+const renderCard = ({ item: listEntry }) => {
+  if (listEntry.origin_key.includes('trip')) {
+    return <TripCard trip={listEntry} />
+  } else if (listEntry.origin_key.includes('place')) {
+    return <PlaceCard place={listEntry} />
+  } else if (listEntry.origin_key.includes('untracked')) {
+    return <UntrackedTimeCard triplike={listEntry} />
+  }
+};
+
+const separator = () => <View style={{ height: 8 }} />
+const bigSpinner = <ActivityIndicator size="large" style={{margin: 15}} />
+const smallSpinner = <ActivityIndicator size="small" style={{margin: 5}} />
+
+const TimelineScrollList = ({ listEntries, loadedRange, pipelineRange, loadMoreFn, isLoading }) => {
+
+  // The way that FlashList inverts the scroll view means we have to reverse the order of items too
+  const reversedListEntries = listEntries ? [...listEntries].reverse() : [];
+
+  const reachedPipelineStart = (loadedRange.start_ts <= pipelineRange.start_ts);
+  const footer =  <LoadMoreButton onPressFn={() => loadMoreFn('past')}
+                                  disabled={reachedPipelineStart}>
+                      { reachedPipelineStart ? "No more travel" : "Show Older Travel"}
+                  </LoadMoreButton>;
+  
+  const reachedPipelineEnd = (loadedRange.end_ts >= pipelineRange.end_ts);
+  const header =  <LoadMoreButton onPressFn={() => loadMoreFn('future')}
+                                  disabled={reachedPipelineEnd}>
+                      { reachedPipelineEnd ? "No more travel" : "Show More Travel"}
+                  </LoadMoreButton>;
+
+  if (isLoading=='replace') {
+    return bigSpinner;
+  } else if (listEntries?.length == 0) {
+    return "No travel to show";
+  } else {
+    return (
+      <FlashList inverted
```

Comment:
I looked around for `inverted` and there is not much documentation for it 
"Reverses the direction of scroll. Uses scale transforms of -1."
Have to try it out to see if it is intuitive.

Reply:
In usage, it feels totally fine. However, if you inspect the HTML and look at what it does, it's pretty gross.

Basically, it uses CSS transforms to flip the entire view along the horizontal axis, like a mirror.
Then, it applies the same transformation to each individual list entry to put them right side up again.

I thought this might be an accessibility concern. As far as a screen reader is concerned, the most recent trips are at the top of the page, not the bottom.
But then again, aren't the most recent trips what a blind / visually impaired user would want to be read first anyway?

Reply:
I did see either an issue or early PR where someone planned to rework FlatList/FlashList to handle inversion "the right way". I don't know if that will come to fruition though, as it seems like a major change.